File: frontend/src/types/reward.ts
export interface Reward {
  id: string;
  name: string;
  description: string;
  coin_cost: number;
  category: string;
  image_url: string;
  is_active: boolean;
  redeemed_quantity?: number;
  is_redeemed?: boolean;
  created_at: string;
}

export interface CoinBalance {
  balance: number;
  total_earned: number;
  total_redeemed: number;
  updated_at: string;
}

export interface RedeemRewardRequest {
  reward_id: string;
  quantity?: number;
}

export interface RedeemRewardResponse {
  redemption_id: string;
  reward_id: string;
  reward_name: string;
  quantity: number;
  unit_cost: number;
  total_cost: number;
  remaining_balance: number;
  redeemed_at: string;
}

File: backend/app/schemas/reward.py
from pydantic import BaseModel, Field, ConfigDict
from datetime import datetime

class RewardRead(BaseModel):
    id: str
    name: str
    description: str
    coin_cost: int
    category: str
    image_url: str
    is_active: bool
    redeemed_quantity: int = 0
    is_redeemed: bool = False
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

class CoinBalanceRead(BaseModel):
    balance: int
    total_earned: int
    total_redeemed: int
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

class RedeemRewardRequest(BaseModel):
    reward_id: str = Field(min_length=1)
    quantity: int = Field(default=1, ge=1)

class RedeemRewardResponse(BaseModel):
    redemption_id: str
    reward_id: str
    reward_name: str
    quantity: int
    unit_cost: int
    total_cost: int
    remaining_balance: int
    redeemed_at: datetime
